### ISSUE TITLE
Fancy Math Rendering

### ISSUE THREAD

**Issue by JasonGross** [HoTT/HoTT#1461]
We now have documentation with [Alectryon](https://plv.csail.mit.edu/blog/alectryon.html).  (TOC [here](https://hott.github.io/HoTT/alectryon-html/toc.html), index [here](https://hott.github.io/HoTT/alectryon-html/), and note that the rest of the pages are much, much prettier than the jank index and TOC)

The blog has some pretty cool examples of fancy rendering that can be done, such as
![image](https://user-images.githubusercontent.com/396076/114895009-1de7dd00-9ddd-11eb-8263-cb5402a402f3.png)
![image](https://user-images.githubusercontent.com/396076/114895057-293b0880-9ddd-11eb-93a8-25d21982a802.png)
![image](https://user-images.githubusercontent.com/396076/114895077-2fc98000-9ddd-11eb-9b20-a9756a7007d7.png)

Are there any files here that would benefit from being rendered or explained with such notations or images or diagrams?  (I have only the vaguest of ideas how to do this, but I'm imagining it might be possible even to get diagram-chasing proofs to render as string diagrams with the right notations.)

**Comment by mikeshulman** [comment]
Absolutely, I'm sure some path-algebra proofs could benefit from some pictures.  (It would be even more useful to be able to see those pictures when *coding* the proofs, but I guess one can't have everything.)

**Comment by JasonGross** [comment]
@cpitclaudel is there any hope of getting the side-by-side view of `alectryon.el`/`rst-mode` to display live-updates of the graphics/notations as we're doing proof editing?

@mikeshulman If you want to point me at some proof in particular and a quick sketch (on paper or in whatever drawing program you like or in LaTeX) of what you think the goal state (and/or the proof script itself) should look like, I'd be happy to take a stab at seeing what sort of graphics we can get.

**Comment by JasonGross** [comment]
Possibly related is the "future work" item on the blog:
> - Integrating with [jsCoq](https://x80.org/rhino-coq/), to allow users to interact with the code directly in the browser (most of the output would be precomputed, but users would also be able to edit the code and recompute the output). For a mock-up of the experience, see [the related tools that I built for F*](https://people.csail.mit.edu/cpitcla/fstar.js/stlc.html).

**Comment by cpitclaudel** [comment]
> @cpitclaudel is there any hope of getting the side-by-side view of alectryon.el/rst-mode to display live-updates of the graphics/notations as we're doing proof editing?

I'd love to do that, in the long run.  Part of figuring that out is determining what kind of figures and rendering methods we want to support.
